lib/ecco/row_type.rb

```diff
@@ -0,0 +1,42 @@
+module Ecco
+  class RowType
+    java_import com.github.shyiko.mysql.binlog.event.EventType
+
+    # MySQL v1 and v2 row events
+    EVENTS = {
+      EventType::WRITE_ROWS      => "WRITE_ROWS",
+      EventType::EXT_WRITE_ROWS  => "WRITE_ROWS",
+      EventType::UPDATE_ROWS     => "UPDATE_ROWS",
+      EventType::EXT_UPDATE_ROWS => "UPDATE_ROWS",
+      EventType::DELETE_ROWS     => "DELETE_ROWS",
+      EventType::EXT_DELETE_ROWS => "DELETE_ROWS"
+    }
+
+    TABLE_EVENT = EventType::TABLE_MAP
+
+    def self.accepted_events
+      EVENTS.keys
+    end
+
+    def self.fetch_event(type)
+      EVENTS.fetch(type)
+    end
+  end
+
+  class RowTypeSave < RowType
+    SAVE_EVENTS = {
+      EventType::QUERY  => "QUERY",
+      EventType::ROTATE => "ROTATE",
+    }
+
+    SAVE_EVENTS = SAVE_EVENTS.merge(EVENTS)
```

Comment:
Not sure overwriting a constant is best practice, if you run with warnings won't it complain?

Reply:
Hmm you're right it did. Should I build `SAVE_EVENTS` exactly like I do `EVENTS` then? I was doing it like this because I tried to limit even more how much duplication is done. It would be kind of neat if I could just include the other hash inside this hash here, but I don't think there is a standard way of doing that?

Reply:
something like this ought to work:

```ruby
HASH = { some: key }.merge(SOME_OTHER_HASH)
```

Reply:
Aha cool! Did that here now! 2b8667dae977854fb6e473ce499e014c25ece474